packages/codecs-core/src/reverseCodec.ts

```diff
@@ -0,0 +1,26 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import { mergeBytes } from './bytes';
+import { Codec } from './codec';
+import { ExpectedFixedSizeCodecError } from './errors';
+
+/**
+ * Reverses the bytes of a fixed-size codec.
+ */
+export function reverseCodec<T, U extends T = T>(codec: Codec<T, U>): Codec<T, U> {
+    const fixedSize = codec.fixedSize;
+    if (fixedSize === null) {
+        throw new ExpectedFixedSizeCodecError('Cannot reverse a codec of variable size.');
+    }
+    return {
+        ...codec,
+        encode: (value: T) => codec.encode(value).reverse(),
+        decode: (bytes: Uint8Array, offset = 0) => {
+            const newBytes = mergeBytes([
+                bytes.slice(0, offset),
+                bytes.slice(offset, offset + fixedSize).reverse(),
+                bytes.slice(offset + fixedSize),
```

Comment:
Don't know if you feel like eliminating the unnecessary copies that happen when:

* `offset` is `0`, or
* `offset + bytes.length === fixedSize`

Reply:
Done in #1630.